react/hooks/useBrowserOffline.js

```diff
@@ -0,0 +1,22 @@
+import { useState, useCallback } from 'react'
+import useEventListener from './useEventListener'
+
+export default function useBrowserOffline() {
+  function isOfflineNow() {
+    return window && window.navigator && window.navigator.onLine === false
+  }
+
+  const [isOffline, setOfflineSince] = useState(isOfflineNow)
```

Comment:
```suggestion
  const [isOffline, setOfflineSince] = useState(isOfflineNow())
```

… at least I think? But then why didn't the test catch that?

Reply:
Yu can pass a function to useState, it will be executed at the first initialization

Reply:
TIL!